IndomitableIbises/Bangazon-Sprint-2
# RC - Department Form, Index.html, Department Templates

## Link to Ticket
References #5 and #6 

## Description of Proposed Changes
- index.html created
- navbar is created
- Department Form Created -- We can now add a department and it saves to the DB and will appear in Department List
- Department Templates for List, Form, and Detail created
- When running server, you should see the index, department list, a single department, and a form to add a department

## Steps to Test

Outline the steps to test

```sh
git fetch --all
git checkout RC-DepartmentForm
code .
```
- Delete **migrations folder** and **db.sqlite3 file**
```
python manage.py makemigrations bang
```
- Check to make sure all tables are created in DB Browser then run server
```
python manage.py runserver
```
- Click on **Departments** button
- You will see placeholder text for Department List View
- Click on the **Add New** button to make a new department
- Enter the new department name and submit
- You will be taken back to Department List View and your newly created department will show




## Impacted Areas in Application

List general components of the application that this PR will affect:

* index.html
* department_list.html
* department_form.html
* department_detail.html
* navbar.html
* department_view.py
* urls.py

## Mentions @username
@hayleylandsberg @sirwin4 @eagobert 

## Definition of Done

1. The project must be fully documented. This includes the following:
   1. Complete README that documents the steps to install the code, how to install any dependencies, or system configuration needed.
   1. Every class must be documented with purpose, author, and methods.
   1. Every method must be documented with purpose and argument list - which itself must contain a short purpose for each argument.
1. The project must be able to run fully, and without errors, on each teammate's system.
1. Fulfills every requirement.
Every line of code has been peer reviewed.
1. For projects that require unit testing, core functionality must be identified and have at least one test for each
